src/js/components/Button/Button.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Button
+ *   https://teamsnap-ui-patterns.netlify.com/patterns/components/button.html
+ *
+ * A common button component that will render the appropriate styles for a button or a button-link. Check out the
+ * teamsnap-ui patterns library for more styling info.
+ *
+ * Example:
+ *  <Button
+ *    text="Click Me"
+ *    handleClick={ () => console.warn('Clicky Clicky') }
+ *    attributes={{ className="Button Button--primary", onFocus: () => console.warn('FOCUSED') }} />
+ *
+ */
+
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import TextLink from '../TextLink'
+import { getClassName } from '../../utils/helpers'
+
+class Button extends PureComponent {
+  getButtonClassName = () => {
+    const { className, color, size, isActive, mods } = this.props
+
+    return getClassName(
+      className,
+      color && `Button--${color}`,
+      size && `Button--${size}`,
+      isActive && "is-active",
+      mods
+    )
+  }
+
+
+  renderButtonLink = () => {
+    const { children, routerLink, location, style, isDisabled, onClick } = this.props
+
+    return (
+      <TextLink
+        className={ this.getButtonClassName() }
+        children={ children }
+        routerLink={ routerLink }
+        location={ location }
+        style={ style }
+        disabled={ isDisabled }
+        onClick={ onClick } />
+    )
+  }
+
+  renderButton = () => {
+    const { children, type, style, isDisabled, onClick } = this.props
+
+    return (
+      <button
+        type={ type }
+        className={ this.getButtonClassName() }
+        style={ style }
+        onClick={ onClick }
+        disabled={ isDisabled }
+        children={ children } />
```

Comment:
Any reason to have children as a prop here? It makes sense for a link that could hold a node, but a button will only hold text.

Reply:
So a button could also hold an Icon.  So we could allow the button to accept `text` and `icon` and even an `iconPosition`, but doing something like children gives full flexibility.  I'm open to a change here though if we would rather.

Reply:
I have seen children as a nested component such as
```
<button> 
  {children}
</button>
```
Does it make any difference passed as a prop?

Reply:
no, I think either one would be accepted as the same 'prop'